src/app/pages/workspace/workspace-search-tile/workspace-search-tile.component.ts

```diff
@@ -0,0 +1,328 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { FilterMetadata, MenuItem, TreeNode } from 'primeng/api';
+import { Table } from 'primeng/table';
+import { Observable, Subject, debounceTime, of, switchMap, throttleTime } from 'rxjs';
+import { ElementType } from 'src/app/models/corpus/element-type';
+import { SearchRequest } from 'src/app/models/search/search-request';
+import { SearchResult, SearchResultRow } from 'src/app/models/search/search-result';
+import { CorpusElement, FolderElement } from 'src/app/models/texto/corpus-element';
+import { CommonService } from 'src/app/services/common.service';
+import { CorpusStateService } from 'src/app/services/corpus-state.service';
+import { LoaderService } from 'src/app/services/loader.service';
+import { SearchService } from 'src/app/services/search.service';
+
+/**selectButton model for search mode*/
+interface SearchMode {
+  name: string,
+  code: string
+  inactive: boolean
+}
+
+@Component({
+  selector: 'app-workspace-search-tile',
+  templateUrl: './workspace-search-tile.component.html',
+  styleUrls: ['./workspace-search-tile.component.scss']
+})
+export class WorkspaceSearchTileComponent implements OnInit {
+
+  constructor(private corpusStateService: CorpusStateService,
+    private searchService: SearchService,
+    private commonService: CommonService,
+    private loaderService: LoaderService) { }
+
+  /**initial panel size */
+  currentPanelHeight: number = 500;
+
+  /**search data parameters */
```

Comment:
Discorso generale: compodoc non prevede i commenti a blocchi e questi sono poco espressivi quando il codice viene ripreso dopo molto tempo, potenzialmente da sviluppatori diversi. Ti prego di commentare in maniera più puntuale perché, per esempio, selectAll alla ln 52 senza un commento è un nome di variabile parlante ma estremamente generico e non sempre si vuole tornare alla definizione delle variabili per capire a cosa si riferiscano.

Reply:
Capisco. Il motivo per cui tendo a non commentare tutte le variabili e in generale il codice, è che noi abbiamo un approccio totalmente all'opposto cioè tendiamo a non commentare nulla, sforzandoci piuttosto di rendere il codice parlante. Questo solo per spiegarti come mai non è un'operazione che mi viene in automatico, ma certamente proverò ad essere più puntuale nei commenti. Il caso specifico selectAll è in effetti poco parlante se visto in mezzo al codice e può essere nominato meglio.

Reply:
Perfetto, grazie